kedro-datasets/kedro_datasets/spark/spark_dataset_v2.py

```diff
@@ -0,0 +1,459 @@
+    DEFAULT_SAVE_ARGS: dict[str, Any] = {}
+
+    def __init__(  # noqa: PLR0913
+        self,
+        *,
+        filepath: str,
+        file_format: str = "parquet",
+        load_args: dict[str, Any] | None = None,
+        save_args: dict[str, Any] | None = None,
+        version: Version | None = None,
+        credentials: dict[str, Any] | None = None,
+        metadata: dict[str, Any] | None = None,
+    ):
+        self.file_format = file_format
+        self.load_args = {**self.DEFAULT_LOAD_ARGS, **(load_args or {})}
+        self.save_args = {**self.DEFAULT_SAVE_ARGS, **(save_args or {})}
+        self.credentials = credentials or {}
+        self.metadata = metadata
+
+        # Parse filepath
+        self.protocol, self.path = self._parse_filepath(filepath)
+
+        # Validate and warn about Databricks paths
+        self._validate_databricks_path(filepath)
+
+        # Get filesystem for metadata operations (exists, glob)
+        exists_function, glob_function = self._get_filesystem_ops()
+
+        # Store Spark compatible path for I/O
+        self._spark_path = self._to_spark_path(filepath)
+
+        # Handle schema if provided
+        self._schema = self._load_schema_from_file(self.load_args.pop("schema", None))
+
+        super().__init__(
+            filepath=PurePosixPath(self.path),
+            version=version,
+            exists_function=exists_function,
+            glob_function=glob_function,
+        )
+
+        self._validate_delta_format()
+
+    def _parse_filepath(self, filepath: str) -> tuple[str, str]:
+        """Parse filepath handling special cases like DBFS."""
+        # Handle DBFS paths
+        if filepath.startswith("/dbfs/"):
+            # /dbfs/path -> dbfs protocol with /path
+            return "dbfs", filepath[6:]  # Remove /dbfs prefix
+        elif filepath.startswith("dbfs:/"):
+            # dbfs:/path -> already in correct format
+            return get_protocol_and_path(filepath)
+        elif filepath.startswith("/Volumes"):
+            # Unity Catalog volumes
+            return "file", filepath
+        else:
+            return get_protocol_and_path(filepath)
+
+    def _validate_databricks_path(self, filepath: str) -> None:
+        """Warn about potential Databricks path issues."""
+        if (
+            deployed_on_databricks()
+            and not (
+                filepath.startswith("/dbfs")
+                or filepath.startswith("dbfs:/")
+                or filepath.startswith("/Volumes")
+            )
+            and not any(
+                filepath.startswith(f"{p}://")
+                for p in ["s3", "s3a", "s3n", "gs", "abfs", "wasbs"]
+            )
+        ):
+            logger.warning(
+                "Using SparkDatasetV2 on Databricks without the `/dbfs/`, `dbfs:/`, or `/Volumes` prefix "
+                "in the filepath is a known source of error. You must add this prefix to %s",
+                filepath,
```

Comment:
As far as I know, Databricks isn't recommending using the `dbfs` file format anymore, so we should make sure this dataset is also up to date with the latest practice on Databricks and probably remove this error message.